taxi/views.py

```diff
@@ -58,8 +60,26 @@ class CarListView(LoginRequiredMixin, generic.ListView):
     queryset = Car.objects.all().select_related("manufacturer")
 
 
-class CarDetailView(LoginRequiredMixin, generic.DetailView):
-    model = Car
+# class CarDetailView(LoginRequiredMixin, generic.DetailView):
+#     model = Car
+
+@login_required()
```

Comment:
@login_required should not have parentheses unless you are passing parameters to it.